src/app/Library/CrudPanel/Traits/Validation.php

```diff
@@ -202,42 +202,35 @@ public function validateRequest()
                 return app($formRequest);
             }
 
-            // create an alias of the provided FormRequest so we can create a new class that extends it.
-            // we can't use $variables to extend classes.
-            class_alias(get_class(new $formRequest), 'DeveloperProvidedFormRequest');
-
-            // create a new anonymous class that will extend the provided developer FormRequest
-            // in this class we will merge the FormRequest rules() and messages() with the ones provided by developer in fields.
-            $extendedRequest = new class($rules, $messages) extends \DeveloperProvidedFormRequest
-            {
-                private $_rules;
-                private $_messages;
-
-                public function __construct($rules, $messages)
-                {
-                    parent::__construct();
-                    $this->_rules = $rules;
-                    $this->_messages = $messages;
-                }
-
-                public function rules()
-                {
-                    return array_merge(parent::rules(), $this->_rules);
-                }
-
-                public function messages()
-                {
-                    return array_merge(parent::messages(), $this->_messages);
-                }
-            };
+            $formRequest = (new $formRequest)->createFrom($this->getRequest());
+            $extendedRules = $this->mergeRules($formRequest, $rules);
+            $extendedMessages = array_merge($messages, $formRequest->messages());
 
             // validate the complete request with FormRequest + controller validation + field validation (our anonymous class)
-            return app(get_class($extendedRequest), ['rules' => $rules, 'messages' => $messages]);
+            return $formRequest->validate($extendedRules, $extendedMessages);
```

Comment:
Because of this, it doesn't work on Demo Monsters.

I'm having a:
`Call to a member function get() on array`

This was previously returning the `$request`, and with the `validate`, it returns an array of validated fields.

Reply:
I get it, I should have used the function I created for this, I will fix this in a sec: 👍 thanks